### PR TITLE
Closes #2388 numeric country decodes

### LINKED ISSUE #2388: Feature Request:  Add numeric decodes for COUNTRY
### Feature Idea

`ADPPK` (Population PK) requires numeric decodes for covariates.  `COUNTRYN` is the numeric decoded version of `COUNTRY`.  Sometimes the list of decodes can be very long for large studies.  One option that some sponsors may use is to use a number based on the ISO 3166-1 alpha-3 codes.

For example:
Set to the country [ADSL.COUNTRY] using the code list which assigns the numerical representation of the standard-defined 249 countries after alphabetical sorting.

1 = ABW (ARUBA)
2 = AFG (AFGHANISTAN)
3 = AGO (ANGOLA)
…

The 249 3-letter codes correspond to the ISO 3166-1 alpha-3 codes:

[https://en.wikipedia.org/wiki/ISO_3166-1_alpha-3 ](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-3 )

The decodes could be stored in function as below or in a tibble.

```
decode_country <- function(x) {
  y <- case_when(
    x == "ABW" ~ 1,
    x == "AFG" ~ 2,
    x == "AGO" ~ 3,
   ...
    x == "ZWE" ~ 249
   )
}
```

### Relevant Input

_No response_

### Relevant Output

_No response_

### Reproducible Example/Pseudo Code

_No response_

### PR DESCRIPTION
**Pull request by jeffreyad**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Review the [Cheat Sheet](https://github.com/pharmaverse/admiral/blob/main/inst/cheatsheet/admiral_cheatsheet.pdf). Make any required updates to it by editing the file `inst/cheatsheet/admiral_cheatsheet.pptx` and re-upload a PDF version of it to the same folder.
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers). A Developer Notes section is available in `NEWS.md` for tracking developer-facing issues.  
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 ## New Features
 
 - Error Messaging has been made more "user-friendly". (#2372)
+- New `country_code_lookup` function added to decode countries based on ISO 3166 codes. (#2388)
```
Comment (manciniedoardo) [inline NEWS.md:6]:
```suggestion
- New `country_code_lookup()` function added to decode countries based on [ISO 3166 codes](https://www.iso.org/iso-3166-country-codes.html). (#2388)
```

File: R/create_country_codes.R [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+#'
+#' @description
+#' These pre-defined country codes are sourced from
+#' [ISO 3166 Standards](https://www.iso.org/iso-3166-country-codes.html).
+#' See also [Wikipedia](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-3).
+#'
+#' @details
+#'
+#' `country_code` is the 3-letter ISO 3166-1 county code commonly found in the
+#' ADSL COUNTRY variable.
```
Comment (manciniedoardo) [inline R/create_country_codes.R:11]:
```suggestion
#' ADSL `COUNTRY` variable.
```

**Review by manciniedoardo — CHANGES_REQUESTED** [review]

**Comment by millerg23** [comment]
I think we have 2 other possible solutions for storing the country codes:
1. Have a function called `get_country_codes` with the tibble inside it, then user can define the name of the dataset or call the function directly ie. `dataset_add = get_country_codes()`.
2. We store the dataset in `data` folder with r script that creates it stored in a sub-folder of `inst` (data.R would need updated also).

I don't think we have any other scripts that creates a dataset directly, like this does. So would be good to be consistent how we create what is essentially metadata?

**Comment by bundfussr** [comment]
> I think we have 2 other possible solutions for storing the country codes:
> 
> 1. Have a function called `get_country_codes` with the tibble inside it, then user can define the name of the dataset or call the function directly ie. `dataset_add = get_country_codes()`.
> 2. We store the dataset in `data` folder with r script that creates it stored in a sub-folder of `inst` (data.R would need updated also).
> 
> I don't think we have any other scripts that creates a dataset directly, like this does. So would be good to be consistent how we create what is essentially metadata?

Usually, datasets are stored in `data` and the scripts which create them should be stored in `data-raw`. Unfortunately, we are not following this convention. I would discuss it at the next meeting.

**Comment by jeffreyad** [comment]
> I think we have 2 other possible solutions for storing the country codes:
> 
> 1. Have a function called `get_country_codes` with the tibble inside it, then user can define the name of the dataset or call the function directly ie. `dataset_add = get_country_codes()`.
> 2. We store the dataset in `data` folder with r script that creates it stored in a sub-folder of `inst` (data.R would need updated also).
> 
> I don't think we have any other scripts that creates a dataset directly, like this does. So would be good to be consistent how we create what is essentially metadata?

Thanks @millerg23 and @bundfussr, I noticed `dose_freq_lookup` in `create_single_dose_dataset()` creates a tibble in this way.  But I agree they should be consistent.

**Review by bms63 — APPROVED** [review]
Looks good to me!

File: NEWS.md [hunk 3]
```diff
@@ -3,6 +3,7 @@
 ## New Features
 
 - Error Messaging has been made more "user-friendly". (#2372)
+- New `country_code_lookup()` function added to decode countries based on [ISO 3166 codes](https://www.iso.org/iso-3166-country-codes.html). (#2388)
```
Comment (bundfussr) [inline NEWS.md:6]:
```suggestion
- New `country_code_lookup()` metadata added to decode countries based on [ISO 3166 codes](https://www.iso.org/iso-3166-country-codes.html). (#2388)
```

File: R/create_country_codes.R [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+#' covar <- adsl %>%
+#'   derive_vars_merged(
+#'     dataset_add = country_code_lookup,
+#'     new_vars = exprs(COUNTRYN = country_number, COUNTRYL = country_name),
+#'     by_vars = exprs(COUNTRY = country_code)
+#'   )
+#' covar
+#'
+#' @rdname country_code_lookup
+country_code_lookup <- tribble(
```
Comment (bundfussr) [inline R/create_country_codes.R:52]: Could we derive `country_number` instead of hard-coding it? Then it is easier to maintain.

Comment (jeffreyad) [reply R/create_country_codes.R:52]: Updated, thanks!

**Review by bundfussr — APPROVED** [review]

File: R/create_country_codes.R [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+#'
+#' @details
+#'
+#' `country_code` is the 3-letter ISO 3166-1 county code commonly found in the
+#' ADSL `COUNTRY` variable.
+#' `country_name` is the country long name corresponding to to the 3-letter code.
+#' `country_number` is the numeric code corresponding to an alphabetic sorting of
+#' the 3-letter codes.
+#'
+#' To see the entire table in the console, run `print(country_code_lookup)`.
```
Comment (bms63) [inline R/create_country_codes.R:16]: nice tip!!

File: R/create_country_codes.R [hunk 6]
```diff
@@ -0,0 +1,307 @@
...
+#'
+#' @family metadata
+#'
+#' @examples
+#' library(tibble)
+#' library(dplyr, warn.conflicts = FALSE)
+#' library(lubridate)
+#'
+#' # Create reference dataset for periods
+#' adsl <- tribble( #' adsl <- tribble(
```
Comment (bms63) [inline R/create_country_codes.R:32]: I think there is a wayward `#'` in here?

**Review by manciniedoardo — APPROVED** [review]

File: R/create_country_codes.R (hunk 6 above)
Comment (bms63) [reply R/create_country_codes.R:32]: @jeffreyad  could you check on this example?  Is the `#'` needed?  after that we can merge

**Pull request merged**